controllers/http.go

```diff
@@ -0,0 +1,85 @@
+package controllers
+
+import (
+	"net/http"
+
+	wssc "github.com/armosec/armoapi-go/apis"
+	"github.com/gin-gonic/gin"
+	"github.com/kubescape/go-logger"
+	"github.com/kubescape/go-logger/helpers"
+	"github.com/kubescape/kubevuln/core/domain"
+	"github.com/kubescape/kubevuln/core/ports"
+)
+
+type HTTPController struct {
+	scanService ports.ScanService
+}
+
+func NewHTTPController(scanService ports.ScanService) *HTTPController {
+	return &HTTPController{
+		scanService: scanService,
+	}
+}
+
+func (h HTTPController) GenerateSBOM(c *gin.Context) {
+	c.Header("Content-Type", "application/json")
+
+	var newScan wssc.WebsocketScanCommand
+	err := c.ShouldBindJSON(&newScan)
+	if err != nil {
+		logger.L().Ctx(c.Request.Context()).Error("handler error", helpers.Error(err))
+		c.JSON(http.StatusBadRequest, nil)
+		return
+	}
+
+	err = h.scanService.GenerateSBOM(c.Request.Context(), newScan.ImageHash, domain.Workload(newScan))
+	if err != nil {
+		logger.L().Ctx(c.Request.Context()).Error("service error", helpers.Error(err))
+		c.JSON(http.StatusInternalServerError, nil)
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{
+		"status": "success",
+		"data":   "new SBOMs created",
```

Comment:
issue: the response does not give its clients a way to see what they have created

Usually, creation APIs return the ID of the created items, their bodies or some kind of a reference to them.

Is there a reason we don’t do something like that? Maybe we should at least give back the ID of the image for which we are creating the SBOM.

I believe this is a new API endpoint that did not exist previously, so we can do it properly.

As a consumer of this API I would like to see a reference to the SBOM that has been created. I know we are currently storing them in memory, so an ID of the SBOM would do. And, actually, it would help even when we store it somewhere else ;)